src/dune_rules/lib.mli

```diff
@@ -160,6 +160,21 @@ module DB : sig
     -> dune_version:Dune_lang.Syntax.Version.t
     -> Compile.t
 
+  (** Resolve libraries written by the user in a [dune] file. The resulting list
+      of libraries is transitively closed and sorted by the order of
+      dependencies.
+
+      This function is for melange.emit stanzas. *)
+  val resolve_user_written_deps_for_melange :
+       t
+    -> (Loc.t * string) list
```

Comment:
This parameter should be:

```
[`Exe of (Loc.t * string) list | `Melange_emit of string]
```

Reply:
https://github.com/ocaml/dune/pull/6454/commits/0b7f398c2184e5109de9055007cf502c9f356198